src/types/app_config_test.go

```diff
@@ -40,6 +40,16 @@ var _ = Describe("AppConfig", func() {
 		})
 	})
 
+	var _ = Describe("NewAppConfig", func() {
+		It("should throw and error if any fields are invalid", func() {
+			appDir := path.Join("..", "..", "example-apps", "invalid-app-config")
+			_, err := types.NewAppConfig(appDir)
+			Expect(err).To(HaveOccurred())
+			expectedErrorString := "only alphanumeric character(s) separated by a single hyphen are allowed. Must match regex: /^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$/"
+			Expect(err.Error()).To(ContainSubstring(expectedErrorString))
+		})
```

Comment:
Can we please have separate tests for an invalid app name and an invalid service name?